src/PrestaShopBundle/Controller/Admin/CommonController.php

```diff
@@ -277,4 +278,43 @@ public function renderFieldAction($formName, $formType, $fieldName, $fieldData)
             'formId' => $formName . '_' . $fieldName . '_rendered',
         ]);
     }
+
+    /**
+     * Process Grid search
+     *
+     * @param Request $request
+     * @param string $gridDefinitionFactoryService
+     * @param string $redirectRoute
+     * @param array $redirectQueryParamsToKeep
+     *
+     * @return RedirectResponse
+     */
+    public function searchGridAction(
+        Request $request,
+        $gridDefinitionFactoryService,
+        $redirectRoute,
+        array $redirectQueryParamsToKeep
+    ) {
+        $definitionFactory = $this->get($gridDefinitionFactoryService);
```

Comment:
all these parameters become required when you only need them because... ?

why is this becoming so complicated? o_O

Reply:
the problem with categories listing is that they are a bit more complicated. 

Category listings are nested, that means once you open categories page you see top most categories and you can "View" each category separately. Once you click "View" it opens new list with sub-categories and parent category ID is available as query parameter.

To make every list possible to filter, sort & paginate, we need to keep some query parameters (in this case parent category id), this is actually pretty common use case, this is why I added additional `redirectQueryParamsToKeep` option.

is it clearer now? :thinking:

Reply:
yes, if it's common in the back office it's ok (this needs to be added somewhere in docs so for people need to migrate or do a grid with the same needs)